Symfony/CS/Fixer/Contrib/MethodArgumentDefaultValueFixer.php

```diff
@@ -0,0 +1,236 @@
+        $nextToken = $tokens[$nextTokenIndex];
+
+        return $nextToken->equals('=');
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return bool
+     */
+    private function isDefaultArgumentAfterIndex(Tokens $tokens, $index)
+    {
+        $nextTokenIndex = $tokens->getNextTokenOfKind($index, $this->immediateDefaultValueTokens);
+        $nextToken = $tokens[$nextTokenIndex];
+
+        return $nextToken->equals('=');
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $variableIndex
+     */
+    private function removeDefaultArgument(Tokens $tokens, $variableIndex)
+    {
+        if ($this->isTypehintedNullableVariable($tokens, $variableIndex)) {
+            return;
+        }
+
+        $argumentEndIndex = $this->findArgumentEndIndex($tokens, $variableIndex);
+        $currentIndex = $tokens->getNextMeaningfulToken($variableIndex);
+
+        while ($currentIndex < $argumentEndIndex) {
+            $tokens[$currentIndex]->clear();
+            $this->clearWhitespacesBeforeIndex($tokens, $currentIndex);
+            $currentIndex = $tokens->getNextMeaningfulToken($currentIndex);
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $variableIndex
+     *
+     * @return bool
+     */
+    private function isTypehintedNullableVariable(Tokens $tokens, $variableIndex)
+    {
+        $prevMeaningfulTokenIndex = $tokens->getPrevTokenOfKind($variableIndex, array(array(T_STRING), ',', '('));
+        $nextMeaningfulTokenIndex = $tokens->getNextTokenOfKind($variableIndex, array(array(T_STRING), ',', ')'));
+
+        $lowerCasedNextContent = strtolower($tokens[$nextMeaningfulTokenIndex]->getContent());
+
+        return
+            $lowerCasedNextContent === 'null' &&
+            $tokens[$prevMeaningfulTokenIndex]->isGivenKind(T_STRING);
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $variableIndex
+     *
+     * @return int
+     */
+    private function findArgumentEndIndex(Tokens $tokens, $variableIndex)
+    {
+        $currentIndex = $variableIndex;
+        while (!$tokens[$currentIndex]->equalsAny($this->argumentTerminatorTokens)) {
+            ++$currentIndex;
+            if ($tokens[$currentIndex]->equals('(')) {
+                $currentIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $currentIndex) + 1;
+            }
+
+            if ($tokens[$currentIndex]->equals('[')) {
+                $currentIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_SQUARE_BRACE, $currentIndex) + 1;
+            }
+        }
+
+        return $currentIndex;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     */
+    private function clearWhitespacesBeforeIndex(Tokens $tokens, $index)
+    {
+        $currentIndex = $index - 1;
+
+        while ($tokens[$currentIndex]->isGivenKind(T_WHITESPACE)) {
```

Comment:
it is not standard to have few `T_WHITESPACE` tokens one after another. document why it is possible here